native_client/c_speech_features/c_speech_features.c

```diff
@@ -13,30 +13,39 @@ csf_mfcc(const short* aSignal, unsigned int aSignalLen, int aSampleRate,
          int aCepLifter, int aAppendEnergy, csf_float* aWinFunc,
          csf_float** aMFCC)
 {
-  int i, j, k, idx, fidx;
+  int i, j, k, idx, fidx, didx;
   csf_float* feat;
   csf_float* energy;
 
   int n_frames = csf_logfbank(aSignal, aSignalLen, aSampleRate, aWinLen, aWinStep,
                               aNFilters, aNFFT, aLowFreq, aHighFreq, aPreemph,
                               aWinFunc, &feat, aAppendEnergy ? &energy : NULL);
 
+  // Allocate an array so we can calculate the inner loop multipliers
+  // in the DCT-II just one time.
+  double* dct2f = (double*)malloc(sizeof(double) * aNFilters * aNCep);
```

Comment:
This is the part that meant that mfcc wasn't performing as well as it should. This didn't affect the optimisation (that was the disabled flags in the BUILD file).